File: backend/micro_epoweredmove_vehicle/src/main/java/com/kosdiam/epoweredmove/models/Vehicle.java
package com.kosdiam.epoweredmove.models;

import java.io.Serializable;

import com.kosdiam.epoweredmove.models.enums.EvType;
import com.kosdiam.epoweredmove.models.enums.VehicleType;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Vehicle implements Serializable {
    private String id;
    private String brand;
    private VehicleType vehicleType;
    private EvType evType;
    private String model;
    private Integer releaseYear;
    private Float usableBatterySize;
    private String plugTypeId;
    private Float avgConsumption;
    private String userId;
    private Long timestamp;
}

File: frontend/E-PoweredMoveAndroidApp/app/src/main/java/com/kosdiam/epoweredmove/models/Vehicle.kt
package com.kosdiam.epoweredmove.models

import android.os.Parcelable
import com.kosdiam.epoweredmove.models.enums.EvType
import com.kosdiam.epoweredmove.models.enums.VehicleType
import kotlinx.android.parcel.Parcelize
import java.util.Calendar

@Parcelize
data class Vehicle(
    var id: String,
    var brand: String,
    var vehicleType: VehicleType,
    var evType: EvType,
    var model: String,
    var releaseYear: Int,
    var usableBatterySize: Float,
    var plugTypeId: String,
    var plugTypeObj: PlugType,
    var avgConsumption: Float,
    var userId: String,
    var userObj: User,
    var timestamp: Long,
) : Parcelable{
    constructor(): this("", "", VehicleType.CAR, EvType.BEV, "", 2023, 1f, "", PlugType(), 1f, "", User(), Calendar.getInstance()
        .timeInMillis)
}
